ownership-chain/precompile/evolution-collection/src/lib.rs

```diff
@@ -169,13 +167,11 @@ where
 	}
 }
 
-impl<AddressMapping, AccountId, TokenUri, LaosEvolution> Precompile
-	for EvolutionCollectionPrecompile<AddressMapping, AccountId, TokenUri, LaosEvolution>
+impl<Runtime> Precompile for EvolutionCollectionPrecompile<Runtime>
```

Comment:
cannot this be removed as in the `evolution-collection-factory` module ?

Reply:
oh, I removed it in `EvolutionCollectionFactory` because there is only one `Action` item to match. Here though, I think it makes sense to move the execution to `inner_execute`